server/src/handlers/influxdb.rs

```diff
@@ -0,0 +1,366 @@
+    Second,
+}
+
+impl Precision {
+    fn normalize(&self, ts: i64) -> i64 {
+        match self {
+            Self::Millisecond => ts,
+            Self::Second => ts * 1000,
+        }
+    }
+}
+
+/// Line protocol
+#[derive(Debug)]
+pub struct WriteRequest {
+    pub lines: String,
+    pub precision: Precision,
+}
+
+impl From<Bytes> for WriteRequest {
+    fn from(bytes: Bytes) -> Self {
+        WriteRequest {
+            lines: String::from_utf8_lossy(&bytes).to_string(),
+            precision: Default::default(),
+        }
+    }
+}
+
+pub type WriteResponse = ();
+
+impl<Q: QueryExecutor + 'static> Influxdb<Q> {
+    pub fn new(instance: InstanceRef<Q>, schema_config_provider: SchemaConfigProviderRef) -> Self {
+        Self {
+            instance,
+            schema_config_provider,
+        }
+    }
+
+    async fn query(
+        &self,
+        ctx: RequestContext,
+        req: QueryRequest,
+    ) -> Result<handlers::query::Response> {
+        handlers::query::handle_query(&ctx, self.instance.clone(), req)
+            .await
+            .map(handlers::query::convert_output)
+    }
+
+    async fn write(&self, ctx: RequestContext, req: WriteRequest) -> Result<WriteResponse> {
+        let request_id = RequestId::next_id();
+        let deadline = ctx.timeout.map(|t| Instant::now() + t);
+        let catalog = &ctx.catalog;
+        self.instance.catalog_manager.default_catalog_name();
+        let schema = &ctx.schema;
+        let schema_config = self
+            .schema_config_provider
+            .schema_config(schema)
+            .box_err()
+            .with_context(|| InfluxdbHandler {
+                msg: format!("get schema config failed, schema:{schema}"),
+            })?;
+
+        let write_context =
+            WriteContext::new(request_id, deadline, catalog.clone(), schema.clone());
+
+        let plans = crate::grpc::storage_service::write::write_request_to_insert_plan(
+            self.instance.clone(),
+            convert_write_request(req)?,
+            schema_config,
+            write_context,
+        )
+        .await
+        .box_err()
+        .with_context(|| InfluxdbHandler {
+            msg: "write request to insert plan",
+        })?;
+
+        let mut success = 0;
+        for insert_plan in plans {
+            success += crate::grpc::storage_service::write::execute_plan(
+                request_id,
+                catalog,
+                schema,
+                self.instance.clone(),
+                insert_plan,
+                deadline,
+            )
+            .await
+            .box_err()
+            .with_context(|| InfluxdbHandler {
+                msg: "execute plan",
+            })?;
+        }
+        debug!(
+            "Remote write finished, catalog:{}, schema:{}, success:{}",
+            catalog, schema, success
+        );
+
+        Ok(())
+    }
+}
+
+fn convert_write_request(req: WriteRequest) -> Result<Vec<WriteTableRequest>> {
```

Comment:
This conversion seems very inefficient. Actually, the underlying write procedure handles the request row by row, so maybe we can convert the write request into `Rows`(defined in common_types/src/row/mod.rs) rather than  `WriteTableRequest` here.

Reply:
> This conversion seems very inefficient. 

Could you explain in more details? All SDK use this `WriteTableRequest` type to do write.

> so maybe we can convert the write request into Rows

Here I reuse write logic for gRPC, and this level reuse is also applied for prom write.  

IMO `Rows` isn't suitable at protocol level. If you want to reuse lower level logic, then more work is needed.

Reply:
> Could you explain in more details? All SDK use this WriteTableRequest type to do write.
Too many clones here. Furthermore, another conversion from the `WriteTableRequest` to `RowGroup` follows, which makes this conversion here redundant.

Reply:
Maybe we can leave the work in the future.

Reply:
I will create a new issue to track this.